python/taichi/misc/gui.py

```diff
@@ -235,6 +240,15 @@ def circles(self, pos, color=0xFFFFFF, radius=1):
             raise ValueError(
                 'Color must be an ndarray or int (e.g., 0x956333)')
 
+        if len(palette) > 0:
+            assert type(palette_indices) != 'NoneType'
+            assert palette_indices.shape == (n, )
+            color_array = np.arange(palette_indices.shape[0])
+            for id in range(palette_indices.shape[0]):
+                color_array[id] = palette[palette_indices[id]]
```

Comment:
We  can convert `palette` to numpy array, so that we can write the following without the loop:

```py
color_array[id] = palette[palette_indices]
```